app/controllers/api/v0/highlights_controller.rb

```diff
@@ -19,32 +21,37 @@ def index
     inbound_binding, error = bind(request.query_parameters, Api::V0::Bindings::GetHighlightsParameters)
     render(json: error, status: error.status_code) and return if error
 
-    query_result = inbound_binding.query(user_uuid: current_user_uuid)
+    query_result = inbound_binding.query(user_id: current_user_uuid)
 
-    response_binding = Api::V0::Bindings::Highlights.create_from_query_result(query_result)
-    render json: response_binding, status: :ok
+    ServiceLimits.get_check(query_result) do |query_result|
+      response_binding = Api::V0::Bindings::Highlights.create_from_query_result(query_result)
+      render json: response_binding, status: :ok
+    end
   end
 
   def update
     inbound_binding, error = bind(params.require(:highlight), Api::V0::Bindings::HighlightUpdate)
     render(json: error, status: error.status_code) and return if error
 
-    model = inbound_binding.update_model!(@highlight)
+    ServiceLimits.update_check(current_user_uuid, inbound_binding) do |binding|
+      model = binding.update_model!(@highlight)
+    end
 
     response_binding = Api::V0::Bindings::Highlight.create_from_model(model)
     render json: response_binding, status: :ok
   end
 
   def destroy
-    @highlight.destroy!
+    ServiceLimits.delete_reset(current_user_uuid, @highlight, &:destroy!)
```

Comment:
don't need to pass in `current_user_uuid` b/c user is available via the passed-in model.  

Do we need to pass in `destroy!` or can we just call it directly?

Maybe a more wordy method name?  `ServiceLimits.destroy_and_reset_counts(@highlight)`